presto-main/src/main/java/io/prestosql/sql/analyzer/StatementAnalyzer.java

```diff
@@ -1879,89 +1875,180 @@ private Scope computeAndAssignOrderByScopeWithAggregation(OrderBy node, Scope so
 
             for (SelectItem item : node.getSelect().getSelectItems()) {
                 if (item instanceof AllColumns) {
-                    analyzeSelectAllColumns((AllColumns) item, node, scope, outputExpressionBuilder);
+                    analyzeSelectAllColumns((AllColumns) item, node, scope, outputExpressionBuilder, selectExpressionBuilder);
                 }
                 else if (item instanceof SingleColumn) {
-                    analyzeSelectSingleColumn((SingleColumn) item, node, scope, outputExpressionBuilder);
+                    analyzeSelectSingleColumn((SingleColumn) item, node, scope, outputExpressionBuilder, selectExpressionBuilder);
                 }
                 else {
                     throw new IllegalArgumentException("Unsupported SelectItem type: " + item.getClass().getName());
                 }
             }
+            analysis.setSelectExpressions(node, selectExpressionBuilder.build());
 
-            List<Expression> result = outputExpressionBuilder.build();
-            analysis.setOutputExpressions(node, result);
-
-            return result;
+            return outputExpressionBuilder.build();
         }
 
         private void analyzeSelectAllColumns(
                 AllColumns allColumns,
                 QuerySpecification node,
                 Scope scope,
-                ImmutableList.Builder<Expression> outputExpressionBuilder)
-        {
-            // expand * and T.*
-            Optional<QualifiedName> starPrefix = allColumns.getTarget()
-                    .map(target -> {
-                        if (target instanceof DereferenceExpression) {
-                            return DereferenceExpression.getQualifiedName((DereferenceExpression) target);
+                ImmutableList.Builder<Expression> outputExpressionBuilder,
+                ImmutableList.Builder<SelectExpression> selectExpressionBuilder)
+        {
+            // expand * and expression.*
+            if (allColumns.getTarget().isPresent()) {
+                // analyze AllColumns with target expression (expression.*)
+                Expression expression = allColumns.getTarget().get();
+
+                // try resolve target expression as table reference or column reference
+                QualifiedName prefix = null;
+                if (expression instanceof DereferenceExpression) {
+                    prefix = DereferenceExpression.getQualifiedName((DereferenceExpression) expression);
+                }
+                else if (expression instanceof Identifier) {
+                    prefix = QualifiedName.of(ImmutableList.of((Identifier) expression));
+                }
+                if (prefix != null) {
+                    // try match prefix to a table
+                    RelationType relationType = scope.getRelationType();
+                    List<Field> fields = relationType.resolveFieldsWithPrefix(Optional.of(prefix));
```

Comment:
This should fail but does not (use memory connector to try it):

```
CREATE SCHEMA a;
USE a;
CREATE TABLE b (g varchar);
CREATE TABLE t (x row(f1 bigint, f2 bigint));

SELECT a.b.* FROM a.b, t AS a;  --- should fail but does not
```

(BTW, this might be a pre-existing bug).